src/main/java/org/blockchain_innovation/factom/identiy/did/parse/FactomMethodSpecificIdRule.java

```diff
@@ -20,8 +21,12 @@ public String execute() throws RuleException {
         if (StringUtils.isEmpty(getDidUrl())) {
             throw new RuleException("A Factom DID cannot have an empty DID scheme");
         }
-
-        String methodSpecificId = FACTOM_V1_JSON.getMethodSpecificId(getDidUrl());
+        String methodSpecificId;
+        try {
+            methodSpecificId = FACTOM_V1_JSON.getMethodSpecificId(getDidUrl());
+        } catch (ParserException e){
+            methodSpecificId = "";
```

Comment:
Why do we allow an invalid method specific Id?

Reply:
Oops, totally missed this. Apparently this was caught by the later `StringUtils.isEmpty` check, but I've updated it to throw with the originating error.